packages/components/src/components/Tooltip/TooltipBox.tsx

```diff
@@ -0,0 +1,109 @@
+import { Icon, IconType } from '../assets/Icon/Icon';
+import { ReactElement, ReactNode } from 'react';
+import { motion } from 'framer-motion';
+import { borders, palette, spacings, spacingsPx, typography } from '@trezor/theme';
+import styled from 'styled-components';
+
+export const TOOLTIP_BORDER_RADIUS = borders.radii.sm;
+
+const getContainerPadding = (isLarge: boolean, isWithHeader: boolean) => {
+    if (isLarge) {
+        if (isWithHeader) {
+            return `${spacingsPx.sm} ${spacingsPx.md} ${spacingsPx.xs}`;
+        }
+
+        return `${spacingsPx.xs} ${spacingsPx.md}`;
+    }
+
+    return spacingsPx.xs;
+};
+
+type TooltipContainerStyledProps = {
+    $maxWidth: string | number;
+    $isLarge: boolean;
+    $isWithHeader: boolean;
+};
+
+const TooltipContainerStyled = styled(motion.div)<TooltipContainerStyledProps>`
+    background: ${palette.darkGray300};
+    color: ${palette.lightWhiteAlpha1000};
+    border-radius: ${TOOLTIP_BORDER_RADIUS};
+    text-align: left;
+    border: solid 1.5px ${palette.darkGray100};
+    max-width: ${props => props.$maxWidth}px;
+    ${typography.hint}
+
+    > div {
+        padding: ${({ $isLarge, $isWithHeader }) => getContainerPadding($isLarge, $isWithHeader)};
+    }
+`;
+
+const HeaderContainer = styled.div`
+    display: flex;
+    align-items: center;
+    justify-content: space-between;
+    gap: ${spacingsPx.sm};
+    width: 100%;
+`;
+
+const TooltipTitle = styled.div<{ $isLarge: boolean }>`
+    display: flex;
+    align-items: center;
+    gap: ${spacingsPx.xxs};
+    width: 100%;
+    color: ${({ theme }) => theme.textSubdued};
+    ${({ $isLarge }) => ($isLarge ? typography.highlight : typography.hint)}
+`;
+
+const Addon = styled.div`
+    margin-left: auto;
+`;
+
+export type TooltipBoxProps = {
+    content: ReactNode;
+    maxWidth?: string | number;
+    /**
+     *  @description Legacy prop
+     */
+    isLarge?: boolean;
+    addon?: ReactNode;
+    headerIcon?: IconType;
+    title?: ReactElement;
+};
+
+type TooltipBoxExtendedProps = TooltipBoxProps &
+    Required<Pick<TooltipBoxProps, 'maxWidth' | 'isLarge'>> & { isOpen: boolean };
+
+export const TooltipBox = ({
+    isOpen,
+    addon,
+    maxWidth,
+    isLarge,
+    content,
+    headerIcon,
+    title,
+}: TooltipBoxExtendedProps) => (
+    <TooltipContainerStyled
+        $isLarge={isLarge}
+        $isWithHeader={!!(title || addon)}
+        $maxWidth={maxWidth}
+        tabIndex={-1}
+        animate={isOpen ? 'shown' : 'hidden'}
```

Comment:
maybe `visible` instead of `shown`?

Reply:
I see it's probably from framer motion...

Reply:
Yes, I know. The naming in the API is not uniform. I kinda struggle between using the original naming or adopting naming from floating-ui. (from their example imlementation) which is:

![image](https://github.com/trezor/trezor-suite/assets/152899911/79ee6dae-90ee-45a7-b5a3-6c2f59047c51)